mlflow/utils/requirements_utils.py

```diff
@@ -0,0 +1,78 @@
+import os
+from itertools import filterfalse
+
+
+def _is_comment(line):
+    return line.startswith("#")
+
+
+def _is_empty(line):
+    return line == ""
+
+
+def _strip_inline_comment(line):
+    return line[: line.find(" #")].rstrip() if " #" in line else line
+
+
+def _is_requirements_file(line):
+    return line.startswith("-r ") or line.startswith("--requirement ")
+
+
+def _join_continued_lines(lines):
+    """
+    Joins lines ending with '\\'.
+
+    >>> _join_continued_lines["a\\", "b\\", "c"]
+    >>> 'abc'
+    """
+    continued_lines = []
+
+    for line in lines:
+        if line.endswith("\\"):
+            continued_lines.append(line.rstrip("\\"))
+        else:
+            continued_lines.append(line)
+            yield "".join(continued_lines)
+            continued_lines.clear()
+
+    # The last line ends with '\'
+    if continued_lines:
+        yield "".join(continued_lines)
+
+# TODO: Add support for constraint files:
+#       https://github.com/mlflow/mlflow/pull/4519#discussion_r668412179
+def _parse_requirements(requirements_file):
+    """
+    A simplified version of `pip._internal.req.parse_requirements` which performs the following
+    operations on the given requirements file and yields the parsed requirements.
```

Comment:
Can we indicate the parameter type for `requirements_file` - i.e. a string path to a requirements file on the local filesystem.

Reply:
Sure!